homeassistant/components/intellifire/config_flow.py

```diff
@@ -40,6 +41,9 @@ def __init__(self):
         """Initialize the Config Flow Handler."""
         self._config_context = {}
         self._not_configured_hosts: list[str] = []
+        self._discovered_hosts: list[
+            tuple(str, str)
+        ] = []  # will store tuples of ip/serial
```

Comment:
You could use a `dict[str, str]` here

Reply:
Or you could make it `self._discovered_host: tuple[str,str] | None = None` since in this case you only expect one

Reply:
You could also use a `namedtuple` or a `dataclass`